scripts/benchmark.py

```
@@ -0,0 +1,515 @@
+    if not data:
+        return 0.0
+    sorted_data = sorted(data)
+    k = (len(sorted_data) - 1) * (p / 100)
+    f = int(k)
+    c = f + 1 if f + 1 < len(sorted_data) else f
+    return sorted_data[f] + (k - f) * (sorted_data[c] - sorted_data[f])
+
+
+def run_benchmark(
+    name: str,
+    func: Any,
```

Ruff (0.14.8)

Dynamically typed expressions (typing.Any) are disallowed in `func`

(ANN401)